File: 0780-max-chunks-to-make-sorted/0780-max-chunks-to-make-sorted.js
/**
 * @param {number[]} arr
 * @return {number}
 */
var maxChunksToSorted = function(arr) {
    let pipe=0;
    let max= -Infinity;
    // if(arr[0]==0)
    // {
    //     pipe++;
    // }

    for(let i=0;i<arr.length;i++)
        {
            if(arr[i]>max)
                max=arr[i];
            if(max===i)
                pipe++;
        }

        return pipe;
};

File: 0780-max-chunks-to-make-sorted/0780-max-chunks-to-make-sorted.cpp
class Solution {
public:
    int maxChunksToSorted(vector<int>& c) {
        int pipe=0;
    int m=INT_MIN;

     for(int i=0;i<c.size();i++)
    {
        m=max(c[i],m);

        if(m==i)
        {
            pipe++;
        }
    }
    return pipe;
    }
};